### PR TITLE
Add BuildLogs method on Client

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

File: pkg/client/buildconfigs.go [hunk 1]
```diff
@@ -37,10 +37,10 @@ func newBuildConfigs(c *Client, namespace string) *buildConfigs {
 }
 
 // List returns a list of buildconfigs that match the label and field selectors.
-func (c *buildConfigs) List(label, field labels.Selector) (result *buildapi.BuildConfigList, err error) {
+func (cfg *buildConfigs) List(label, field labels.Selector) (result *buildapi.BuildConfigList, err error) {
```
Comment (smarterclayton) [inline pkg/client/buildconfigs.go]: Shorter type reference variables are usually better.  Why is cfg better than c?  This is more of a client interface and a cfg interface.

Comment (kargakis) [inline pkg/client/buildconfigs.go]: Got it. Rolled back all the renaming

**Comment by smarterclayton** [comment]
Rebase

**Comment by kargakis** [comment]
Done

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by kargakis** [comment]
Why did this hang?

**Comment by smarterclayton** [comment]
This

```
a08fe19f456db0bfed57b88d60f94a08b65ab5b0195fd2c7335b77c026341eeb: unable to sethostname "build-ruby-sample-build-validtag-f28818d5-c2ac-11e4-894d-22000b339afe": invalid argument
```
is the too long hostname issue which is https://github.com/GoogleCloudPlatform/kubernetes/issues/4825

**Comment by smarterclayton** [comment]
trying merge again [merge]

**Pull request merged**